index.js

```diff
@@ -170,7 +172,7 @@ module.exports = function (h, opts) {
           state = TEXT
         } else if (state === COMMENT && /-$/.test(reg) && c === '-') {
           if (opts.comments) {
-            res.push([ATTR_VALUE,reg.substr(0, reg.length - 1)],[CLOSE])
+            res.push([ATTR_VALUE,reg.substr(0, reg.length - 1)])
```

Comment:
the [CLOSE] tag appears to have been unnecessary, because one is inserted in the next iteration (when the character is `>`)